testing/upload_validation.ts

```diff
@@ -1367,6 +1369,24 @@ const packMetadataSchemaBySdkVersion: SchemaExtension[] = [
 
           const schemaPathPrefix = ['syncTables', i, 'schema'];
           validateSchemaDeprecatedFields(syncTable.schema, schemaPathPrefix, context);
+
+          // A blank string will fail the existing `min(1)` requirement in the zod schema, so
+          // we only need to additionally make sure there is a value present.
+          if (typeof syncTable.identityName !== 'string') {
+            let receivedType: ZodParsedType = ZodParsedType.unknown;
+            if (syncTable.identityName === undefined) {
+              receivedType = ZodParsedType.undefined;
+            } else if (syncTable.identityName === null) {
+              receivedType = ZodParsedType.null;
+            }
```

Comment:
Hmm, I guess it's fine to give you the extra info that you passed undefined or null, but I think `expected` and `received` are optional, so this could all maybe be simplified to this I think?

```typescript
if (!syncTable.identityName) {
  context.addIssue({
    code: z.ZodIssueCode.invalid_type,
    path: ['syncTables', i, 'identityName'],
    message: 'An identityName is required on all sync tables',
  });
}
```

Reply:
Ah I see, we've been using `ZodIssueCode.custom` for all of these similar custom validations, which seems fine here too if you want.